Backend/users/usersControllers.js

```diff
@@ -0,0 +1,71 @@
+require("dotenv").config();
+const bcrypt = require("bcryptjs");
+const User = require("./usersModels");
+const { generateToken } = require("./usersHelper");
+
+exports.signup = async (req, res) => {
+  try {
+    const { email, username, password } = req.body;
+    const hashedPassword = await bcrypt.hash(password, 12);
+    const credentials = {
+      username,
+      email,
+      password: hashedPassword,
+    };
+    const emailExist = await User.findBy({
+      email: credentials.email,
+    });
+    const usernameExist = await User.findBy({
+      username: credentials.username,
+    });
+    if (emailExist || usernameExist) {
+      if (emailExist)
+        res.status(409).json({
+          errorMessage: "Oops, email already exists",
```

Comment:
We should not show if email or username exists because it let you know that account exists, even if you don't know the password yet. It should show `Unknown credentials` or stuff like that

Reply:
check